applications/serial_lte_modem/doc/TCPIP_AT_commands.rst

```diff
@@ -0,0 +1,1541 @@
+  * ``4`` - OMA TLV
+
+* The ``<size>`` value is an integer and it represents the actual number of bytes received.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#UDPRECVFROM="test.server.com",1234
+   Test OK
+   #XRECVFROM: 1, 7
+   OK
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command is not supported.
+
+Resolve hostname #XGETADDRINFO
+==============================
+
+The ``#XGETADDRINFO`` command allows you to resolve hostnames to IPv4 addresses.
+
+Set command
+-----------
+
+The set command allows you to resolve hostnames to IPv4 addresses.
+
+Syntax
+~~~~~~
+
+::
+
+   #XGETADDRINFO=<hostname>
+
+The ``<hostname>`` parameter is a string.
+It cannot be an IPv4 address string.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XGETADDRINFO=<ip_addr>
+
+* The ``<ip_addr>`` value is a string and it indicates the IPv4 address of the resolved hostname.
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xgetaddrinfo="www.google.com"
+   #XGETADDRINFO: 172.217.174.100
+   OK
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command is not supported.
+
+TCP server #XTCPSVR
+===================
+
+The ``#XTCPSVR`` command allows you to start and stop the TCP server.
+
+Set command
+-----------
+
+The set command allows you to start and stop the TCP server.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPSVR=<op>[<port>[,<sec_tag>]]
+
+
+* The ``<op>`` parameter can accept one the following values:
+
+  * ``0`` - Stop server
+  * ``1`` - Start server
+  * ``2`` - Start server with data mode support
+
+* The ``<port>`` parameter is an integer and it represents the TCP service port.
+  It is mandatory for starting the server.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: <handle> started
+
+The ``<handle>`` value is an integer.
+When positive, it indicates that it opened successfully.
+When negative, it indicates that it failed to open.
+
+Unsolicited notification
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: <error> stopped
+
+The ``<error>`` value is a negative integer, and it represents the error value according to the standard POSIX errorno.
+
+::
+
+   #XTCPDATA: <datatype>, <size>
+
+* The ``<datatype>`` value can assume one the following values:
+
+  * ``0`` - hexidecimal string (e.g. "DEADBEEF" for 0xDEADBEEF)
+  * ``1`` - plain text (default value)
+  * ``2`` - JSON
+  * ``3`` - HTML
+  * ``4`` - OMA TLV
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xtcpsvr=1,3442
+   #XTCPSVR: 2 started
+   OK
+   #XTCPSVR: 5.189.130.26 connected
+   #XTCPRECV: 1, 13
+   Hello, TCP#1!
+   #XTCPRECV: 1, 13
+   Hello, TCP#2!
+
+Read command
+------------
+
+The read command allows you to check the TCP server settings.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTCPSVR?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTCPSVR: <listen_socket_handle>,<income_socket_handle>,<data_mode>
+
+The ``<handle>`` value is an integer.
+When positive, it indicates that it opened successfully.
```

Comment:
Please verify if this description is actually correct.

Reply:
It is, socket handle returned by BSDLib is a positive integer